airflow/models/baseoperator.py

```diff
@@ -546,7 +546,8 @@ def dag(self, dag):
                 "The DAG assigned to {} can not be changed.".format(self))
         elif self.task_id not in dag.task_dict:
             dag.add_task(self)
-
+        elif self.task_id in dag.task_dict and dag.task_dict[self.task_id] != self:
```

Comment:
Do we want `dag.task_dict[self.task_id] != self`, or `dag.task_dict[self.task_id] is not self`.

The former use `__eq__`, the later checks for exactly the same object.

Reply:
Updated, I will update Airflow Master as we currently raise errors after `==` check

https://github.com/kaxil/airflow/blob/c3a46b9fec618489314a4ac8d4be289481e8edd3/airflow/models/baseoperator.py#L603-L606